akka-stream/src/main/scala/akka/stream/io/Framing.scala

```diff
@@ -0,0 +1,173 @@
+package akka.stream.io
+
+import java.nio.ByteOrder
+
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ TerminationDirective, PushPullStage, SyncDirective, Context }
+import akka.util.{ ByteIterator, ByteStringBuilder, ByteString }
+
+import scala.annotation.tailrec
+
+object Framing {
+
+  def delimiter(delimiterBytes: ByteString, maximumFrameLength: Int): Flow[ByteString, ByteString, Unit] =
+    Flow[ByteString].transform(() ⇒ new DelimiterFramingStage(delimiterBytes, maximumFrameLength))
+      .named("delimiterFraming")
+
+  def lines(delimiter: String, maximumLineBytes: Int): Flow[ByteString, String, Unit] =
```

Comment:
btw, they are of course pretty much the same :) This is just a convenience method to avoid all the boilerplate of creating ByteStrings, or converting them to Strings.